src/main/flight/imu.c

```diff
@@ -529,3 +526,73 @@ void imuSetHasNewData(uint32_t dt)
 }
 #endif
 
+void imuQuaternionComputeProducts(quaternion *quat) {
```

Comment:
These quaternion products are ugly - computing rotation matrix makes sense, because it is as fast as quaternion rotation and has 'physical' meaning.
Caching product will provide minor speed gain (multiplication is fast on FPU, possibly faster than memory access) and this is only micro-optimization